src/components/Contacts/ContactsContext/ContactsContext.tsx

```diff
@@ -257,8 +257,15 @@ export const ContactsProvider: React.FC<ContactsContextProps> = ({
     _: React.MouseEvent<HTMLElement>,
     view: string,
   ) => {
-    setViewMode(view as TableViewModeEnum);
-    updateOptions(view as TableViewModeEnum);
+    const newViewMode = view as TableViewModeEnum;
+    saveContactsView(newViewMode);
+    if (newViewMode === TableViewModeEnum.Map) {
+      // When switching to the map, make the filter only show the selected contacts, if any
+      setActiveFilters({ ids });
+    } else if (viewMode === TableViewModeEnum.Map) {
+      // When switching away from the map, reset the filter to show all contacts
+      setActiveFilters({});
```

Comment:
I wonder if there is a way to save the filters so a user doesn't lose them if they switch to the map and then back to list view.

One thought I had is to store them in local storage

Reply:
I need to test more to make sure that I didn't break anything else, but this should be fixed now. The contacts wrapper still stores the contact filters in memory, but it only uses the `ids` filter in map view. If you switch back to the list view, it still remembers the previous filters and restores them.